web/ui/src/pages/PageComponentList.tsx

```diff
@@ -1,15 +1,33 @@
 import { faCubes } from '@fortawesome/free-solid-svg-icons';
 
 import ComponentList from '../features/component/ComponentList';
+import { SortOrder } from '../features/component/types';
 import Page from '../features/layout/Page';
 import { useComponentInfo } from '../hooks/componentInfo';
 
 function PageComponentList() {
-  const components = useComponentInfo('');
+  const [components, setComponents] = useComponentInfo('');
+
+  // TODO: make this sorting logic reusable
+  const handleSorting = (sortField: string, sortOrder: SortOrder): void => {
+    if (!sortField || !sortOrder) return;
+    const sorted = [...components].sort((a, b) => {
+      const sortValueA = sortField === 'Health' ? a.health.state.toString() : a.localID;
+      const sortValueB = sortField === 'Health' ? b.health.state.toString() : b.localID;
+      if (sortValueA === null) return 1;
+      if (sortValueB === null) return -1;
+      return (
+        sortValueA.localeCompare(sortValueB, 'en', {
+          numeric: true,
+        }) * (sortOrder === SortOrder.ASC ? 1 : -1)
+      );
+    });
+    setComponents(sorted);
+  };
```

Comment:
To make the sorting easier to extend you could use a map:

```
const fieldMappings: { [key: string]: (comp: any) => string | undefined } = {
    Health: comp => comp.health?.state?.toString(),
    localID: comp => comp.localID,
   // Add new fields if needed here.
};

function getSortValue(component: any, field: string): string | undefined {
    const valueGetter = fieldMappings[field];
    return valueGetter ? valueGetter(component) : undefined;
}
```

And you could use it as follow in the sorting function:
(sortValueA and sortValueB can now be 'undefined', that's why I changed the if statement as well)
```
    const sortValueA = getSortValue(a, sortField);
    const sortValueB = getSortValue(b, sortField);

    if (!sortValueA) return 1;
    if (!sortValueB) return -1;
```

Reply:
Done this one too in latest commit.